app/controllers/exports_controller.rb

```diff
@@ -43,16 +44,28 @@ def locate_labware
   def locate_ancestor_plate
     return nil if export.ancestor_purpose.blank?
 
-    # Store the result of the query in order to support multiple ancestor plates
-    # for the specified ancestor purpose.
-    @ancestor_plate_list = @plate.ancestors.where(purpose_name: export.ancestor_purpose)
-
-    ancestor_result = @ancestor_plate_list.first
+    ancestor_result = @plate.ancestors.where(purpose_name: export.ancestor_purpose).first
     return nil if ancestor_result.blank?
 
     Sequencescape::Api::V2.plate_with_custom_includes(include_parameters, id: ancestor_result.id)
   end
 
+  # Returns an array of all ancestor plates of @plate that match a specific
+  # purpose (defined in the export configuration). If no such plates are found,
+  # an empty array is returned. The result array is made available to the view
+  # that generates the export as @ancestor_plate_list by the show method.
+  #
+  # @return [Array, Sequencescape::Api::V2::Plate] An array of Plate records if
+  #   any are found, otherwise an empty array.
+  def locate_ancestor_plate_list
+    return nil if export.ancestor_purpose.blank?
```

Comment:
should that return empty array instead of nil to be consistent?

Reply:
Thanks for spotting that. Yes, it should return an empty array as described in the method documentation.